### PR TITLE
Optimize sparse softmax cross entropy with logits

### PR DESCRIPTION
**Pull request by zhijxu-MS**
if label is one-hot, some optimization can be done to avoid un-necessary computation.

### CODE REVIEW

File: tf2onnx/onnx_opset/nn.py [hunk 1]
```diff
@@ -706,6 +706,33 @@ def version_7(cls, ctx, node, **kwargs):
         _make_softmax_cross_entropy_with_logits(ctx, labels, logits, node)
 
 
+def _make_sparse_softmax_cross_entropy_with_logits(ctx, label, logit, tf_ori_node):
```
Comment (nbcsm) [inline tf2onnx/onnx_opset/nn.py]: does this work for SoftmaxCrossEntropyWithLogits?

Comment (zhijxu-MS) [reply tf2onnx/onnx_opset/nn.py]: not.  SoftmaxCrossEntropyWithLogits's input "label" **not needs to be one-hot** in tensorflow then the math equation is not equal.

https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits_v2?hl=en
